linalg/ginkgo.cpp

```diff
@@ -25,113 +25,314 @@
 namespace mfem
 {
 
-namespace GinkgoWrappers
+namespace Ginkgo
 {
 
-GinkgoIterativeSolverBase::GinkgoIterativeSolverBase(
-   const std::string &exec_type, int print_iter, int max_num_iter,
-   double RTOLERANCE, double ATOLERANCE)
-   : exec_type(exec_type),
-     print_lvl(print_iter),
-     max_iter(max_num_iter),
-     rel_tol(sqrt(RTOLERANCE)),
-     abs_tol(sqrt(ATOLERANCE))
+GinkgoExecutor::GinkgoExecutor(ExecType exec_type)
 {
-   if (exec_type == "reference")
+   switch (exec_type)
    {
-      executor = gko::ReferenceExecutor::create();
+      case GinkgoExecutor::REFERENCE:
+      {
+         executor = gko::ReferenceExecutor::create();
+         break;
+      }
+      case GinkgoExecutor::OMP:
+      {
+         executor = gko::OmpExecutor::create();
+         break;
+      }
+      case GinkgoExecutor::CUDA:
+      {
+         if (gko::CudaExecutor::get_num_devices() > 0)
+         {
+            executor = gko::CudaExecutor::create(0, gko::OmpExecutor::create());
+         }
+         else
+            mfem::err << "gko::CudaExecutor::get_num_devices() did not report "
+                      << "any valid devices" << std::endl;
```

Comment:
Are you saying the above lines (52-53) are connected to the segfault at 447, or are they two separate issues? 
I'm assuming they are two separate issues since the default device should be `cpu` and so the `CudaExecutor` code shouldn't be involved...?  

I do think it probably would be a good idea to make it an abort rather than just an error message.  I will make that change.

I don't get a segfault with default options with my build setup.  Could you provide a backtrace?